### ISSUE TITLE
Is there a sniff to format array key => value pairs?

### ISSUE THREAD

**Issue by akalongman** [squizlabs/PHP_CodeSniffer#3514]
Is there a sniff to format and check array key => value pairs? Like this:
![Peek 2021-12-20 16-16](https://user-images.githubusercontent.com/423050/146769315-51261ab7-b182-4c9e-83fc-87c30d67446f.gif)

**Comment by jrfnl** [comment]
You can try using something like the below for that:
```xml
<rule ref="Squiz.Arrays.ArrayDeclaration">
    <exclude name="Squiz.Arrays.ArrayDeclaration.KeyNotAligned"/>
    <exclude name="Squiz.Arrays.ArrayDeclaration.ValueNotAligned"/>
    <exclude name="Squiz.Arrays.ArrayDeclaration.CloseBraceNotAligned"/>
    <exclude name="Squiz.Arrays.ArrayDeclaration.CloseBraceNewLine"/>
</rule>
<rule ref="Generic.Arrays.ArrayIndent"/>
```

**Comment by akalongman** [comment]
@jrfnl thank you for the answer, but this configuration complains on arrays like this:

```php
    'default' => env('DB_CONNECTION', 'mysql'),

    'connections' => [
        . . . 
    ]
```

```
20 | ERROR | [x] Array double arrow not aligned correctly; expected 5 space(s) but found 1
     |       |     (Squiz.Arrays.ArrayDeclaration.DoubleArrowNotAligned)
```

If there is an empty lines between keys, should not be aligned

**Comment by jrfnl** [comment]
> If there is an empty lines between keys, should not be aligned

Well, this sniff is of the opinion they should be.

I know the [WordPressCS](https://github.com/WordPress/WordPress-Coding-Standards/) external standard has some [more configurable](https://github.com/WordPress/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#array-alignment-dealing-with-multi-line-items) [array formatting sniffs](https://github.com/WordPress/WordPress-Coding-Standards/tree/develop/WordPress/Sniffs/Arrays).
The external [PHPCSExtra](https://github.com/phpcsstandards/phpcsextra#normalizedarrays) standard will also - in time - contain a full set of highly configurable array formatting sniffs.

You may want to have a look at those.

**Comment by gsherwood** [comment]
There are no other array formatting rules than the ones @jrfnl has pointed out. There are no rules to look at whitespace and change alignment based on that.

I'm going to close this as the question has been answered. Might be worth using a discussion for this in the future.
